make_queue/static/make_queue/js/disable_stream_name_field.js

```diff
@@ -0,0 +1,7 @@
+$(document).ready(function () {
+    $("select[name=machine_type]").change(function () { 
+        var input = document.getElementById("id_stream_name");
+        input.disabled = this.value != 1;
```

Comment:
Ideally this should read the `has_stream` property of `machine_type`, but to get it to dynamically change as the choicefield changes, I had to do it in javascript. And I don't think it is possible to get hold of the `has_stream` property in javascript?

Reply:
Fixed